test/distributed/org/apache/cassandra/distributed/test/metric/TableMetricTest.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.test.metric;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.InstanceNotFoundException;
+import javax.management.ObjectName;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.distributed.api.IInvokableInstance;
+import org.apache.cassandra.distributed.test.TestBaseImpl;
+import org.apache.cassandra.utils.MBeanWrapper;
+
+import static org.apache.cassandra.config.CassandraRelevantProperties.IS_DISABLED_MBEAN_REGISTRATION;
+import static org.apache.cassandra.config.CassandraRelevantProperties.MBEAN_REGISTRATION_CLASS;
+
+public class TableMetricTest extends TestBaseImpl
+{
+    static
+    {
+        MBEAN_REGISTRATION_CLASS.setString(MapMBeanWrapper.class.getName());
+        IS_DISABLED_MBEAN_REGISTRATION.setBoolean(false);
+    }
+
+    private static Map<String, Collection<String>> SYSTEM_TABLES = ImmutableMap.of(
+    "system", ImmutableSet.of("available_ranges", "available_ranges_v2", "batches", "built_views", "compaction_history", "IndexInfo", "local",
```

Comment:
Could you use `org.apache.cassandra.db.SystemKeyspace#ALL_TABLE_NAMES` here? and similarly create ALL_TABLE_NAMES for the other keyspaces where they're missing?

Reply:
I didn't want to as that would load the whole DB, I could refactor to avoid that but felt simplest to just fork.

Reply:
I don't see that in 3.0 or trunk...

Reply:
My mistake :)

Reply:
I think it is better to load the tables programmatically. So compiler prompts when code changes. It can be done via

```java
        Map<String, Collection<String>> systemTables = new HashMap<>();
        Consumer<KeyspaceMetadata> loader = meta ->
        {
            Set<String> tables = meta.tables.stream().map(t -> t.name).collect(Collectors.toSet());
            systemTables.put(meta.name, tables);
        };
        Arrays.asList(SystemKeyspace.metadata(), AuthKeyspace.metadata(), SystemDistributedKeyspace.metadata(),
                      SchemaKeyspace.metadata(), TraceKeyspace.metadata())
              .forEach(loader);
```

Reply:
that has the same issue, we load the whole DB into the app space, so was trying to avoid that.

Reply:
I switched to doing this, I do this by loading all tables in the first instance.